scalding-core/src/main/scala/com/twitter/scalding/serialization/PositionInputStream.scala

```diff
@@ -27,20 +27,23 @@ object PositionInputStream {
 
 class PositionInputStream(val wraps: InputStream) extends InputStream {
   private[this] var pos: Long = 0L
-  private[this] var markPos: Long = -1L
   def position: Long = pos
 
   override def available = wraps.available
+
   override def close() { wraps.close() }
+
   override def mark(limit: Int) {
-    wraps.mark(limit)
-    markPos = pos
+    sys.error("Mark and reset are not supported on this stream")
```

Comment:
The thinking here was that using mark and reset in passed around things was unsafe, and the code I had that used it would break in a meta sense. So making the inner input streams not support it allowed me to hint that only the outermost ones should use it. That said I'm not sure in the current vamp its actually a required change anymore... generally the approach I have at the moment looks like::
```
 def compareBinary($inputStreamA: _root_.java.io.InputStream, $inputStreamB: _root_.java.io.InputStream): _root_.com.twitter.scalding.serialization.OrderedSerialization.Result = {
          try {

            val $lenA = ${readLength(inputStreamA)}
            require($lenA >= 0, "Length was " + $lenA + "which is < 0, invalid")
            val $lenB = ${readLength(inputStreamB)}
            require($lenB >= 0, "Length was " + $lenB + "which is < 0, invalid")


            val subSetRes = if(($lenA == $lenB) && $lenA > 24 && $inputStreamA.markSupported && $inputStreamB.markSupported) {
              $inputStreamA.mark($lenA)
              $inputStreamB.mark($lenB)

              var pos = 0
              var isSame = true

              while(pos < $lenA && isSame == true) {
                $inputStreamA.readByte == $inputStreamB.readByte
                pos = pos + 1
              }

              if(isSame) {
                0
              } else {
                // rewind if they don't match for doing the full compare
                $inputStreamA.reset()
                $inputStreamA.reset()
                -1
              }
            } else {
              -1
            }

            val r = if(subSetRes == 0) {
              0
              } else {
              val $bufferedStreamA = _root_.com.twitter.scalding.serialization.PositionInputStream($inputStreamA)
              val initialPositionA = $bufferedStreamA.position
              val $bufferedStreamB = _root_.com.twitter.scalding.serialization.PositionInputStream($inputStreamB)
              val initialPositionB = $bufferedStreamB.position

              val innerR = ${t.compareBinary(bufferedStreamA, bufferedStreamB)}

              $bufferedStreamA.seekToPosition(initialPositionA + $lenA)
              $bufferedStreamB.seekToPosition(initialPositionB + $lenB)
              innerR
            }

            _root_.com.twitter.scalding.serialization.OrderedSerialization.resultFrom(r)
            }
            catch { case _root_.scala.util.control.NonFatal(e) =>
              _root_.com.twitter.scalding.serialization.OrderedSerialization.CompareFailure(e)
            }
          }

```